### PR TITLE
Update dist_training_gpu.rst

### PR DESCRIPTION
**Pull request by ForFishes**
Modify distributed GPU training excellent practice documents. It uses the Fleet API interface.
image
image
image
image
image
image
image

### CODE REVIEW

**Comment by zq19** [comment]
Please add the PR to http://wiki.baidu.com/pages/viewpage.action?pageId=917662867

File: doc/fluid/advanced_usage/best_practice/dist_training_gpu.rst [hunk 1]
```diff
@@ -1,142 +1,96 @@
...
-如果需要强制使用TCP方式通信，可以设置 :code:`NCCL_IB_DISABLE=1` 。
-
+3、数据准备
+===========
 
-使用GPU完成部分图片预处理
-++++++++++++++++++++++
+1、使用GPU完成部分图片预处理
 
 如果可能，使用GPU完成可以部分数据预处理，比如图片Tensor的归一化：
```
Comment (gavin1332) [inline doc/fluid/advanced_usage/best_practice/dist_training_gpu.rst]: 使用GPU可以完成部分数据预处理？可以使用GPU完成部分数据预处理？

File: doc/fluid/advanced_usage/best_practice/dist_training_gpu.rst [hunk 2]
```diff
@@ -152,63 +106,14 @@ PaddlePaddle Fluid使用“线程池” [#]_ 模型调度并执行Op，Op在启
...
-      pyreader.decorate_tensor_provider(
-        batch_feeder(
-            paddle.batch(rd, batch_size=batch_size_per_gpu),
-            pin_memory=True,
-            img_dtype='uint8'
-        )
-      )
-
-4. 减少reader初始化时间 (infinite read）
+ - 使用 :code:`pyreader` 。参考 `这里 <../../user_guides/howto/prepare_data/use_py_reader.html>`_ 使用pyreader，并开启 :code:`use_double_buffer` 。
```
Comment (gavin1332) [inline doc/fluid/advanced_usage/best_practice/dist_training_gpu.rst:115]: 链接失效

File: doc/fluid/advanced_usage/best_practice/dist_training_gpu.rst [hunk 3]
```diff
@@ -1,142 +1,96 @@
...
+       dist_optimizer = fleet.distributed_optimizer(optimizer, strategy=dist_strategy)
+       dist_optimizer.minimize(avg_cost)
+                ...
+   for pass_id in range(PASS_NUM):
        batch_id = 0
        while True:
-           if batch_id % 30 == 0:
-               fetched = exe.run(fetch_list)
+           if batch_id % fetch_frequency == 0:        #fetch频率
+               fetched = exe.run(main_prog, fetch_list)
```
Comment (gavin1332) [inline doc/fluid/advanced_usage/best_practice/dist_training_gpu.rst:86]: code missing in “VIEW FILE”

Comment (ForFishes) [reply doc/fluid/advanced_usage/best_practice/dist_training_gpu.rst:86]: Done, I will take a screenshot for this file.

File: doc/fluid/advanced_usage/best_practice/dist_training_gpu.rst (hunk 2 above)
Comment (ForFishes) [reply doc/fluid/advanced_usage/best_practice/dist_training_gpu.rst:115]: Done

File: doc/fluid/advanced_usage/best_practice/dist_training_gpu.rst (hunk 1 above)
Comment (ForFishes) [reply doc/fluid/advanced_usage/best_practice/dist_training_gpu.rst]: Done

**Review by gavin1332 — APPROVED** [review]
LGTM

**Review by hutuxian — APPROVED** [review]

**Review by gavin1332 — APPROVED** [review]

**Pull request merged**